src/boot/idle.c

```diff
@@ -52,9 +53,9 @@ void Main_InitMemory(void) {
     void* memStart = (void*)0x80000400;
     void* memEnd = OS_PHYSICAL_TO_K0(osMemSize);
 
-    Main_ClearMemory(memStart, gFramebuffer1);
-    Main_ClearMemory(D_80025D00, bootproc);
-    Main_ClearMemory(gGfxSPTaskYieldBuffer, memEnd);
+    Main_ClearMemory(memStart, SEGMENT_START(framebuffer_1));
+    Main_ClearMemory(SEGMENT_START(pre_boot_buffer), SEGMENT_START(boot));
+    Main_ClearMemory(SEGMENT_END(code), memEnd);
```

Comment:
Kinda would have expected something like (pseudocode)

```suggestion
    Main_ClearMemory(memStart, SEGMENT_START(framebuffer_1));
    Main_ClearMemory(SEGMENT_START(pre_boot_buffer), SEGMENT_END(pre_boot_buffer));
    Main_ClearMemory(SEGMENT_START(heap), memEnd);
```
, doesn't match? Or is this not what this effectively does?

Reply:
also, shouldn't `Main_ClearMemory(memStart, SEGMENT_START(framebuffer_1));` not be hardcoded to end at the start of `framebuffer_1`?

Reply:
I'm not sure what the first clear does but maybe (again idk if matches)

Reply:
I'll answer in the order of the function calls

As a reference, this is on main right now:
```c
    Main_ClearMemory(memStart, gFramebuffer1);
    Main_ClearMemory(D_80025D00, bootproc);
    Main_ClearMemory(gGfxSPTaskYieldBuffer, memEnd);
```
https://github.com/zeldaret/mm/blob/71d83a5e15c69e5f2e91f45f55448f49bb215897/src/boot/idle.c#L55C45-L57

- `Main_ClearMemory(memStart, SEGMENT_START(framebuffer_1));`

Is clearing the range `0x80000400 ~ 0x80000500`. I have no idea what's in that range or why it is clearing it up.
We don't have any symbol that points to that range either. `gFramebuffer1` happens to be at `0x80000500` so we were using that.
I thought it would be better to use a linker symbol for this so I changed it (I don't think it is correct either, but less wrong than what we had).

- `Main_ClearMemory(SEGMENT_START(pre_boot_buffer), SEGMENT_START(boot));`

Sadly this can't reference the end of `pre_boot_buffer` because the way I set it up.
That segment only has 1 symbol that has size 1, so the end of the segment is at `start(pre_boot_buffer) + 0x10`, which isn't the address that we want to point to, that's why I used `SEGMENT_START(boot)` instead.
If we really wanted to change this to refer to the end of this segment, then we could change the size of the only symbol it has to have size `0x5A360`, that should bump the end linker symbol up to were we need it.

- `Main_ClearMemory(SEGMENT_END(code), memEnd);`

This one clears all the ram after `code`, since all of that is assumed to usable by the game as buffers. This PR is in a weird position because now there are 3 segments after the `code` one, so referring to only one of them feels a bit weird.
From all the 3 I think this one is the only one that would match by simply changing the symbol to `SEGMENT_START(buffers)`

Reply:
Just remembered when I worked on the z_syscfb pr, I came up with these notes on how the buffers are allocated: https://github.com/zeldaret/mm/pull/1159#discussion_r1118077166

pre_boot_buffer is phony, the "framebuffer_1" segment should actually be u16 [HIRES_BUFFER_HEIGHT][HIRES_BUFFER_WIDTH] (454 x 576), which will get you to 0x80080000, then the missing 0x60 bytes is the space reserved for the makerom entrypoint.

So the second Main_ClearMemory is wiping `framebuffer_1` and the makerom entrypoint, only it's skipping over the lores color framebuffer being stored there for w/e reason

Reply:
This is interesting.
The second `Main_ClearMemory` only wipes `pre_boot_buffer` segment, it doesn't touch the `framebuffer_1` segment.
`gFramebufferHiRes1` (which is placed at the same address of `framebuffer_1`) is bigger than the `framebuffer_1` segment and it leaks into `pre_boot_buffer`, so this call technically wipes a chunk of `gFramebufferHiRes1`, but it doesn't touch `gFramebuffer1` at all

Reply:
And now I understand why the lores buffer is avoided; the memory at 80000500 to 80025D00 has already initialized by `Main_InitScreen`.